### DISCUSSION TITLE
Sequencing multiple releases

### DISCUSSION THREAD

**Discussion by harshavmb** [goreleaser/goreleaser#3472] [Ideas, Issue Triage, and General discussions]
Hi,

We are publishing artifacts to artifactory & then create metadata using custom publishers. For custom publishers, the artifacts have to exist (remotely) in first place. 

```
artifactories:
  - name: production
    target: http://<Your-Instance>:8081/artifactory/example-repo-local/{{ .ProjectName }}/{{ .Version }}/
    username: "user"
    checksum: true
    signature: true
    
publishers:
  - name: Create metadata & publish
    cmd: >-
      /tmp/tf-provider-registry-api-generator      
      --bucket-name 'abc'
      --terraform-provider '{{ .ProjectName }}'      
      --namespace 'xyz'
      --fingerprint '{{ .Env.GPG_FINGERPRINT }}'
      --provider-version {{ .Tag }}
```

Nevertheless, I'm seeing `publishers` running ahead of `artifactories` or both at the same time.

How can I sequence `publishers` to execute only after `artifactories`? I've checked documentation, couldn't find any.

**Comment by caarlos0** [comment]
custom publishers are always running before artifactory... although I don't see any particular reason for it... 

that said, we don't have the notion of user-defined ordering in goreleaser... it static.

The order of publishing is defined in internal/pipe/publish/publish.go

pretty much everything before `release.Pipe{}` on line 45 could be reordered. I think we can pub the custom publishers before the docker stuff.

**Reply by caarlos0** [reply]
https://github.com/goreleaser/goreleaser/pull/3473

**Comment by harshavmb** [comment]
> How can I sequence `publishers` to execute only after `artifactories`? I've checked the documentation, but couldn't find any.

Many Thanks. Much appreciated for such a quick reply & a PR!

I guess someone else using `goreleaser` might be doing exactly the opposite of what I'm doing. In such cases, wouldn't this be a problem? I just want to be sure that my requirements won't break others' flows.

**Reply by caarlos0** [reply]
it might, but all in all I think running the custom publishers last makes more sense...

if that happens and someone reports it, I might need to revert it...

**Reply by harshavmb** [reply]
Thanks for the merge. I've tested `1.12.1` today, worked as expected. 
Thanks again!
